### PR TITLE
change addContact to addTicket

### PR DESCRIPTION
**Pull request by neerajdotname**

### CODE REVIEW

**Comment by neerajdotname** [comment]
@breezyboa _a

**Comment by neerajdotname** [comment]
@sudeeptarlekar I'm trying to make `name` optional.

However when I do not pass `name` or pass `name` as empty string then ticket record is not being recorded. Do you have any idea what the fix is https://monosnap.com/file/lviUruKIb9Amyy8rOiPUnWTBmuRBTy

**Comment by breezyboa** [comment]
@neerajdotname since tickets are created against an article. Shouldn't the addTicket mutation take in an article id also as an argument to be associated with an article? 
Also currently in this mutation, the email field is mandatory. According to the requirement, the email field should be optional.

**Comment by neerajdotname** [comment]
You are right. It should take article_id.

Email is required for a ticket. If email is not filled then client needs to call addFeedback and not addTicket.

**Comment by breezyboa** [comment]
changed from addContacts to addTicket in elm
Note that name is also considered as a mandatory field. This should be an optional field in this mutation

once addFeedback mutation is ready, will use it when email is not provided

**Comment by breezyboa** [comment]
@neerajdotname _a

**Comment by neerajdotname** [comment]
@breezyboa probably you missed this comment of mine https://github.com/bigbinary/acehelp/pull/192#issuecomment-404559188

**Comment by breezyboa** [comment]
Oh. yes sorry my bad

**Comment by vipulnsward** [comment]
@breezyboa _a All this is still using old code and old-routes: https://monosnap.com/file/v14GAlJgeeRAaRLFgYqNdbP5ZoD1Rb

Please fix the requests on this page as well as the names

**Comment by breezyboa** [comment]
@vipulnsward _a 
changed from contact api to addTicket graphql mutation in contact us screen

File: app/controllers/api/v1/ticket_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class TicketController < BaseController
```
Comment (vipulnsward) [inline app/controllers/api/v1/ticket_controller.rb]: @sudeeptarlekar _a can you check if this is redundant now?

**Comment by vipulnsward** [comment]
@prathamesh-sonpatki _a

File: app/javascript/Client/Request/Article.elm [hunk 2]
```diff
@@ -41,7 +42,13 @@ requestSearchArticles searchTerm =
...
 
 
--- GRAPHQL
+-- requestFeedbackMutation : FeedbackForm -> Reader ( NodeEnv, ApiKey ) (Task GQLClient.Error (Maybe (List GQLError)))
+-- requestFeedbackMutation feedbackFrom =
+--     Reader.Reader
+--         (\( env, apiKey ) ->
+--             GQLClient.sendMutation (graphqlUrl env) <|
+--                 GQLBuilder.request feedbackFrom feedbackMutation
+--         )
```
Comment (vipulnsward) [inline app/javascript/Client/Request/Article.elm]: Lets get rid of this

**Pull request merged**